Message Card

LiAustin059 opened the pull request:
Added a message card function that takes title, message, and author and outputs it like this.
Not finalized  some issues there, just a base of what I have that works atm
![image](https://user-images.githubusercontent.com/44851962/122628332-85dbcd80-d083-11eb-9b28-9501a1359165.png)

closes #19

LiAustin059 committed:
Create MessageCard.module.scss

LiAustin059 committed:
Made base message card

Message card itself works just working on making the list and it being two on a page instead of just one

LiAustin059 committed:
Ignore last file change about this

Was testing some stuff trying to figure out how this works regard last commit about this. These commits are just so I break something I have a somewhat stable version to use.

LiAustin059 committed:
Update MessageCard.module.scss

LiAustin059 committed:
Update variables.scss

LiAustin059 committed:
Update MessageCard.tsx

RedMaps commented:
Just saying, you might wanna merge main into this soon, looks like its way behind rn ^^

LiAustin059 committed:
s

LiAustin059 committed:
Merge branch 'main' into MessageCard

LiAustin059 committed:
wip

LiAustin059 commented:
Will do thanks for the feedback, ignore last three commits trying to pull from main, trying to figure this out.

LiAustin059 committed:
Made the message card list

Message card list should work just need to adjust grid sizing and container to fit on the grid.

LiAustin059 commented:
Will push from my main branch after this commit, just figuring things out.

LiAustin059 committed:
Merge branch 'main' into MessageCard

LiAustin059 committed:
Stable Build

Just need to play around with the grid settings and should work.

LiAustin059 committed:
Merge branch 'MessageCard' of https://github.com/Austinx88/kaichou-graduation into MessageCard

LiAustin059 committed:
Update MessageCard.module.scss

LiAustin059 committed:
This was a testing file to test elements

LiAustin059 committed:
Update messageProps.ts

LiAustin059 committed:
Delete variables.scss

LiAustin059 commented:
Here it looks like so far:
![image](https://user-images.githubusercontent.com/44851962/123028653-b6867480-d3ad-11eb-84b8-e713c063eea6.png)
Ignore the deleting of variable file as I thought it would delete it from commit not in total.

RedMaps commented:
This seems to have multiple build errors right now. I get this when i start dev: 
![image](https://user-images.githubusercontent.com/16868419/123033746-890aec80-d3e8-11eb-886c-e5ba52746750.png)
Also just from the pictures the spacing of the text seems to be off quite a bit as well as the thing at the top right being a bit too big.

LiAustin059 committed:
Revert "Delete variables.scss"

This reverts commit b1cdde2c18de9ed0b5fd77c49efa3431c5e913de.

LiAustin059 committed:
Made the page for the messages

LiAustin059 committed:
Update index.tsx

LiAustin059 committed:
Fixed some issues about certain elements

LiAustin059 commented:
Sorry about that, thanks for the feedback.  
![image](https://user-images.githubusercontent.com/44851962/123155934-b1213c80-d436-11eb-9b08-eae77a0e901b.png)
Here is how it looks like currently on desktop.

nyxordinal commented:
![image](https://user-images.githubusercontent.com/85892079/123351964-834c0d00-d588-11eb-9dc2-20e1cfeb775c.png)
Is it supposed to be like this? or the two columns should be in the center of the screen?

nyxordinal commented:
Also when I try to build the frontend an error still pops up
![image](https://user-images.githubusercontent.com/85892079/123352823-57318b80-d58a-11eb-8bb8-a8f9541fe527.png)

LiAustin059 committed:
Fixed Compile issue

LiAustin059 committed:
Moved to Message Cards folder

LiAustin059 committed:
Moved to message cards folder

LiAustin059 committed:
Adjusted the grid should look better

LiAustin059 commented:
Fixed the current issues however found out that it doesn't look as wanted with empty space on text and on mobile doesn't look proper as well so working on that

LiAustin059 committed:
Update MessageCard.module.scss

LiAustin059 committed:
Update MessageList.tsx

LiAustin059 committed:
Update package.json

LiAustin059 commented:
Finished the message card page, hopefully last commit. 
![image](https://user-images.githubusercontent.com/44851962/123498711-b1206880-d5ff-11eb-93b9-0406adb661b4.png)
Something to note is that react stack grid needs to be install in order to run, not sure if something does it automatically once the package.json file is changed.

nyxordinal commented on package.json:
```diff
@@ -38,6 +38,7 @@
     "lerna": "^4.0.0",
     "prettier": "2.3.1",
     "typescript": "^4.3.2"
```
``
```suggestion
    "typescript": "^4.3.2",
    "react-stack-grid": "^0.7.1"
```

nyxordinal replied:
minor typo in package.json

nyxordinal commented:
![image](https://user-images.githubusercontent.com/85892079/123514137-8a276c80-d6bb-11eb-85fd-e593d3439459.png)
Sorry, this is what it looks like when I try to run it on my local machine, am I missing somewhere?

azuraga commented:
I have the same inconsistency with card spacing.

azuraga commented:
> I have the same inconsistency with card spacing. 

Update on the spacing: it seems to happen upon first fresh load of the page, and gets resolved upon reload and resizing the browser window.

azuraga commented on packages/frontend/components/MessageCards/MessageCard.module.scss:
```diff
@@ -0,0 +1,58 @@
+  }
+
+  .message {
+    font-size: 24;
```
Missing unit. I think you mean `px`, but that seems to be a bit too big compared to what you had before. `16px` works fine.

azuraga reviewed (changes requested):
I suggested a thing about the `font-size` unit. Also, the fonts don't have a `serif`, `sans-serif`, etc. attached, which, together with the `font-size` thing, might have caused some undefined behavior.

Also, isn't it better to just have an array of card data, instead of an array of *JSONs* containing card data?

azuraga commented:
Since the spacing issue has to do with animation, the optimized production build might automatically resolve it.

LiAustin059 committed:
Update package.json

Fixed typo

Co-authored-by: Nyxordinal <85892079+nyxordinal@users.noreply.github.com>

LiAustin059 committed:
Fixed the font sizing issue.

LiAustin059 commented:
> I suggested a thing about the `font-size` unit. Also, the fonts don't have a `serif`, `sans-serif`, etc. attached, which, together with the `font-size` thing, might have caused some undefined behavior.
> 
> Also, isn't it better to just have an array of card data, instead of an array of _JSONs_ containing card data?

Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.

azuraga commented on package.json:
```diff
@@ -37,7 +37,9 @@
-    "typescript": "^4.3.2"
+    "typescript": "^4.3.2",
+    "react-stack-grid": "^0.7.1"
+    "react-stack-grid": "^0.7.1",
```
```suggestion

```

azuraga reviewed (commented):
> Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.

How's progress on this?

LiAustin059 committed:
Changed to take in an array of card data

LiAustin059 committed:
Update package.json

Co-authored-by: azuraga <73472335+azuraga@users.noreply.github.com>

LiAustin059 commented:
> > Now that I think of it, it does make more sense to do instead. First I thought I needed a key thing with id to work but looking back it does nothing, will change it.
> 
> How's progress on this?

Just finished it, sorry about being late was busy all day. Free in the next couple of days can work more whenever needed.

azuraga commented on packages/frontend/pages/messages/index.tsx:
```diff
@@ -1,11 +1,55 @@
   return (
     <>
       <Navigation title="Messages" page={Page.MESSAGES} />
+      <MessageList key={'1'} cardData={Messages} />
```
Key isn't necessary in this case.
```suggestion
      <MessageList cardData={Messages} />
```

azuraga reviewed (changes requested):
> sorry about being late was busy all day

No worries! I have one small change request, and then I think it's ready to be merged.

LiAustin059 committed:
Update packages/frontend/pages/messages/index.tsx

Co-authored-by: azuraga <73472335+azuraga@users.noreply.github.com>

LiAustin059 committed:
Update messageProps.ts

nyxordinal commented:
Hi! It seems that the gap bug still exists when the page is first loaded, maybe you want to consider using `masonry-board` as in #73 ?

LiAustin059 commented:
Will do, I'll get working on that

LiAustin059 committed:
Merge branch 'Kaichou-Project:main' into MessageCard

LiAustin059 committed:
Used a masonry board system instead

LiAustin059 committed:
Based on previous change

LiAustin059 commented:
Messages page works now on fresh load and looks like this 
![image](https://user-images.githubusercontent.com/44851962/123847760-f46f2580-d8e4-11eb-8329-cbf14b0fbeb8.png)

LiAustin059 committed:
Fixed small issue

azuraga commented:
Doesn't work properly on mobile. I'm looking for a fix
![image](https://user-images.githubusercontent.com/73472335/123858996-6c107500-d924-11eb-9947-e6b3f9c49c16.png)

azuraga commented:
> Doesn't work properly on mobile. I'm looking for a fix
> ![image](https://user-images.githubusercontent.com/73472335/123858996-6c107500-d924-11eb-9947-e6b3f9c49c16.png)

update: fix lays in modifying `styles.container`

azuraga commented:
It might be better to inherit the Fanart MasonryBoard instead of creating a new one that barely has any difference. With that we can also easily fix bugs like this.

LiAustin059 commented:
Will do, at first thought it was better to make it two separate things to make it easier to make small changes.

azuraga commented:
> Will do, at first thought it was better to make it two separate things to make it easier to make small changes.

Indeed, but since the two are pretty much the same concept, we could add those modifications in a different manner. I've incorporated using the fanart board for messages, too, will suggest the code in a minute.

azuraga commented on packages/frontend/components/MessageCards/MessageList.tsx:
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import MessageCard from './MessageCard'
+import MessageProps from '../../interfaces/messageProps'
+import styles from './MessageCard.module.scss'
+import MasonryBoard from './MasonryBoard'
+
+export default function MessageList(props: MessageProps) {
+  const cards = props.cardData.map((message, i) => (
+    <MessageCard key={i} {...message} />
+  ))
+  return (
+    <div className={styles.container}>
+      <MasonryBoard> {cards}</MasonryBoard>
+    </div>
+  )
+}
```
```suggestion
import React from 'react'
import MessageCard from './MessageCard'
import MessageProps from '../../interfaces/messageProps'
import styles from '../Fanart/styles.module.scss'
import MasonryBoard from '../MasonryBoard'

export default function MessageList(props: MessageProps) {
  const cards = props.cardData.map((message, i) => (
    <MessageCard key={i} {...message} />
  ))
  return (
    <div className={styles.fanart_board}>
      <MasonryBoard> {cards}</MasonryBoard>
    </div>
  )
}
```

azuraga reviewed (changes requested):
Fix for the message card misalignment. Will use 3 messages per row, can be adjusted once `MasonryBoard` is properly implemented. This will be done in a future PR.

LiAustin059 committed:
Update packages/frontend/components/MessageCards/MessageList.tsx

Co-authored-by: azuraga <73472335+azuraga@users.noreply.github.com>

azuraga commented:
Alright one small thing. The one difference between fanart and messages in that board seems to be the breakpoints, so can you add a `breakpointCols` parameter to `MasonryBoard` and have `FanartBoard` and `MessageBoard` use appropriate parameters? For `FanartBoard` you can copy the `breakpointColsObj` from `MasonryBoard`, for `MessageBoard` the following seems to work:

```js
const breakpointColumnsObj = {
  default: 3,
  1160: 2,
  860: 1,
}
```

Also, the unused MasonryBoard stuff in the MessageCard can be removed.

LiAustin059 commented:
Sure shouldn't take too long

azuraga committed:
suggested changes

LiAustin059 committed:
Update MessageList.tsx

azuraga committed:
Merge remote-tracking branch 'Austinx88/MessageCard' into PR45fork

# Conflicts:
#	packages/frontend/components/MessageCards/MessageList.tsx

LiAustin059 committed:
Update index.tsx

LiAustin059 committed:
Update FanartBoard.tsx

LiAustin059 committed:
Update FanartBoard.tsx

LiAustin059 committed:
Delete MasonryBoard.tsx

LiAustin059 committed:
Update MessageList.tsx

LiAustin059 committed:
Delete MasonryBoard.module.scss

azuraga committed:
naming consistency, removed fanart breakpoints since these align with the default values

azuraga committed:
Merge remote-tracking branch 'Austinx88/MessageCard' into PR45fork

# Conflicts:
#	packages/frontend/components/MasonryBoard/index.tsx
#	packages/frontend/components/MessageCards/MessageList.tsx

nyxordinal commented:
Please try to merge the latest commit from main branch again into this PR

LiAustin059 committed:
Merge branch 'Kaichou-Project:main' into MessageCard
